File: NetFramework/Schedule/VB/CustomForms/Locations.vb
Imports System
Imports System.Collections.Generic
Imports System.Text

Namespace TestSchedule
	''' <summary>
    ''' This class represents an Exercise location (for example, Studio or Pool).
	''' </summary>
	Class Location
		Private _id As Integer
		Private _name As String

		''' <summary>
		''' Creates new instance of Location object whith the specified parameters.
		''' </summary>
		''' <param name="id"></param>
		''' <param name="name"></param>
		Public Sub New(ByVal id As Integer, ByVal name As String)
			_id = id
			_name = name
		End Sub

		''' <summary>
		''' Gets or sets Location name.
		''' </summary>
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set
				If _name <> value Then
					_name = value
				End If
			End Set
		End Property

		''' <summary>
		''' Unique Id of Location.
		''' </summary>
		Public Property Id() As Integer
			Get
				Return _id
			End Get
			Set
				If _id <> value Then
					_id = value
				End If
			End Set
		End Property

		''' <summary>
		''' Returns LocationName.
		''' </summary>
		''' <returns></returns>
		Public Overloads Overrides Function ToString() As String
			Return Name
		End Function
	End Class
End Namespace

File: NetFramework/Schedule/CS/CustomData/Locations.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestSchedule
{
	/// <summary>
	/// This class represents an Exercise location (for example, Studio or Pool).
	/// </summary>
	class Location
	{
		private int _id;
		private string _name;

		/// <summary>
		/// Creates new instance of Location object whith the specified parameters.
		/// </summary>
		/// <param name="id"></param>
		/// <param name="name"></param>
		public Location(int id, String name) 
		{
			_id = id;
			_name = name;
		}

		/// <summary>
		/// Gets or sets Location name.
		/// </summary>
		public string Name
		{
			get
			{
				return _name;
			}
			set
			{
				if (_name != value)
				{
					_name = value;
                }
			}
		}

		/// <summary>
		/// Unique Id of Location.
		/// </summary>
		public int Id
		{
			get
			{
				return _id;
			}
			set
			{
				if (_id != value)
				{
					_id = value;
				}
			}
		}

		/// <summary>
		/// Returns LocationName.
		/// </summary>
		/// <returns></returns>
		public override string ToString()
		{
			return Name;
		}
	}
}
